src/ucp/wireup/wireup.c

```diff
@@ -1622,6 +1612,25 @@ ucp_wireup_try_select_lanes(ucp_ep_h ep, unsigned ep_init_flags,
     return UCS_OK;
 }
 
+static void
+ucp_wireup_gather_pending_reqs(ucp_ep_h ep,
+                               ucs_queue_head_t *replay_pending_queue)
+{
+    ucs_queue_iter_t iter;
+    ucp_request_t *req;
+
+    ucp_wireup_eps_pending_extract(ep, replay_pending_queue);
+
+    ucs_queue_for_each_safe(req, iter, &ep->worker->rkey_ptr_reqs,
+                            send.rndv.rkey_ptr.queue_elem) {
+        if (req->send.ep == ep) {
+            ucs_queue_push(replay_pending_queue,
+                           (ucs_queue_elem_t*)&req->send.uct.priv);
+            ucs_queue_del_iter(replay_pending_queue, iter);
+        }
+    }
```

Comment:
what happens if we don't do this?
i would expect rkey_ptr protocol reset function will remove these requests anyway? during ucp_proto_rndv_rkey_ptr_reset

Reply:
right, so we can remove `ucs_queue_del_iter`, but still we need to push the request to `replay_pending_queue`